src/tribler/core/utilities/tiny_tribler_service.py

```diff
@@ -97,8 +99,9 @@ async def _terminate_by_timeout(self):
 
     def _graceful_shutdown(self):
         self.logger.info("Shutdown gracefully")
-        task = asyncio.create_task(self.session.shutdown())
-        task.add_done_callback(lambda result: self._stop_event_loop())
+        tasks = self.async_group.add(self.session.shutdown())
```

Comment:
A side question unrelated to this PR: do we need to pass multiple coroutines to the `async_group.add` method? It may be easier to handle the result if we only pass a single coroutine on each call of `async_group.add`:
```python
shutdown_task = self.async_group.add(self.session.shutdown())
```
I did a quick search, and it seems we never pass multiple coros to `async_group.add`

Reply:
nitpick: Another side note, unrelated to the current PR: it may be good to rename the `AsyncGroup.add` method to `AsyncGroup.add_task`. This way, we can find all places where it is called. Right now, it is hard to search for all places where `AsyncGroup.add` is called because the `add` name is too generic.

Reply:
Will fix it in the next PR.